examples/use_case_project_manager/composio_tool.py

```diff
@@ -0,0 +1,147 @@
+from typing import Literal, Optional
+
+from composio import Action, ComposioToolSet
+from pydantic import BaseModel, ConfigDict, Field
+
+from dynamiq.nodes import NodeGroup
+from dynamiq.nodes.agents.exceptions import ToolExecutionException
+from dynamiq.nodes.node import Node, ensure_config
+from dynamiq.runnables import RunnableConfig
+from dynamiq.utils.logger import logger
+
+
+def generate_pydantic_model(schema_dict: dict, model_name: str = "GeneratedModel") -> type[BaseModel]:
+    """
+    Generates a Pydantic BaseModel class from a schema dictionary.
+
+    Args:
+        schema_dict (dict): Dictionary containing field definitions
+        model_name (str): Name for the generated model class
+
+    Returns:
+        type[BaseModel]: Generated Pydantic model class
+    """
+    annotations = {}
+    field_definitions = {}
+
+    type_mapping = {"string": str, "integer": int, "number": float, "boolean": bool, "array": list}
+
+    for field_name, field_info in schema_dict.items():
+        field_type = type_mapping.get(field_info.get("type", "string"), str)
+
+        # Handle arrays
+        if field_type == list and "items" in field_info:
+            item_type = type_mapping.get(field_info["items"].get("type", "string"), str)
+            field_type = list[item_type]
+
+        # Create Field with description and default if present
+        field_kwargs = {
+            "description": field_info.get("description", ""),
+        }
+
+        if "default" in field_info:
+            field_kwargs["default"] = field_info["default"]
+            annotations[field_name] = Optional[field_type]
+        else:
+            annotations[field_name] = field_type
+
+        field_definitions[field_name] = Field(**field_kwargs)
+
+    # Create the namespace for the class
+    namespace = {"__annotations__": annotations, **field_definitions}
+
+    # Create the model class
+    return type(model_name, (BaseModel,), namespace)
+
+
+class ComposioTool(Node):
```

Comment:
Should we move this tool from the examples section to the tools section?

Reply:
We will keep this tool in our separate repository along with the dynamiq tools. This arrangement is temporary and is intended for article publication.